modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -54,16 +59,29 @@ protected void configure() {
             waitingFor(
                 new WaitAllStrategy()
                     .withStrategy(waitStrategy)
-                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
+                    .withStrategy(createLogWaitingStrategy())
```

Comment:
If instead of conditionally switching to the `WaitAllStrategy` we would just use a `WaitAllStrategy` as default and conditionally _add_ the `LogMessageWaitStrategy` to it, it should simplify the implementation (no need for overriding `withStartupTimeout` anymore).

Reply:
@kiview i don't know what i should do here. :( Can you explain that a little more? :)

Reply:
I was thinking something like this in the constructor:
```
waitingFor(
            new WaitAllStrategy()
                .withStrategy(Wait.forHttp(METRICS_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT))
        );
```

And this in configure:
```
waitingFor(
                new WaitAllStrategy()
                    .withStrategy(waitStrategy)
                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1)
                        // NOTE withStartupTimeout must be called at the end, because start timeout will be
                        // applied to all strategies above
                    .withStartupTimeout(startupTimeout))
            );
```

But think somewhere along with the commits in this PR, we broke the `shouldWaitForFunctionsWorkerStarted` test. It fails on CI and locally now. Maybe I gave a wrong suggestion, we have to look into this.